### PR TITLE
Freeze teacher parameters in Distillation and DistillationV2

### PR DESCRIPTION
**Pull request by OshriFatkiev**
## What has changed and why?

- Freeze the teacher parameters in both `Distillation` (v1) and `DistillationV2` by
  setting `requires_grad=False` on all teacher parameters in `get_teacher`.
- This avoids counting the teacher parameters as "trainable" even though the teacher
  is in eval mode and not optimized, and prevents accidental gradient computation
  through the teacher.
- Added tests for both `Distillation` and `DistillationV2` to assert that all
  teacher parameters have `requires_grad == False`.

## How has it been tested?

- `pytest tests/_methods/distillation/test_distillation.py`
- `pytest tests/_methods/distillationv2/test_distillationv2.py`

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes  
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes  
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

**Comment by guarin** [comment]
Thanks for the contribution, this will probably also remove the pytorch lightning warning which is a nice bonus:
```
Found 212 module(s) in eval mode at the start of training. This may lead to unexpected behavior during training. If this is intentional, you can ignore this warning.
```

**Review by guarin — APPROVED** [review]
